### DISCUSSION TITLE
Debugging Cloud (Listenbrainz Token) functionality

### DISCUSSION THREAD

**Discussion by john-** [jcorporation/myMPD#1053] [Q&A] [answered]
myMPD version: 10.3.1

I have configured myMPD with Listenbrainz token and checked the option "Show MusicBrainz links".  However, after a song is fully played LIstenbrainz Web site indciates that nothing has been listened to.

1. Generate Listenbrainz token on the Listenbrainz Web siite.
2. Configure myMPD with Listenbrainz token
3. Play a song
4. Go to Listenbrainz Web site
5. See that nothing has been listened to

I have enabled myMPD debugging but do not see anything obviously related to communicating with Listenbrainz.  I have only tried this with myMPD in jukebox mode.

[mympd_debug.txt.gz](https://github.com/jcorporation/myMPD/files/11899511/mympd_debug.txt.gz)

**Comment by jcorporation** [comment, marked as answer]
I thought I had written a documentation for this, but I can not find anything...

Scrobbling is not buildin in the core of myMPD. Scrobbling works through myMPD scripting with lua and a trigger. There is a tested script in the repository that can be easily imported.

1. Create a scrobbler script
   1. Main menu -> Scripts -> New
   2. Import "ListenBrainz-Scrobbler.lua" (This script uses the Listenbrainz token from the settings dialog)
2. Add a Scrobble trigger referencing above script

Edit: found the documentation, it is somewhat hidden: https://jcorporation.github.io/myMPD/scripting/usage/scrobble-to-listenbrainz

**Reply by chiefy** [reply]
Got here after putting in my LB token and waiting for something to happen :) Might be good to put a little `*` in the UI [that links to that doc](https://jcorporation.github.io/myMPD/additional-topics/listenbrainz)? (link above is not working)

**Comment by john-** [comment]
Thanks!  I will give that a try and report back.  Would implementing those steps also show the MusicBrainz links in the UI or does that require additional configuration?

**Reply by jcorporation** [reply]
> Would implementing those steps also show the MusicBrainz links in the UI or does that require additional configuration?

Enabling MusicBrainz tags in MPD and myMPD and checking the option "Show MusicBrainz links" is enough.

**Comment by john-** [comment]
I enabled the MusicBrainz tags and get the links in the UI.

As far as Musicbrainz scrobbling I imported the script and created the trigger for it.  Songs still do not appear in ListenBrainz Web site.

[mympd_debug.txt.gz](https://github.com/jcorporation/myMPD/files/11906731/mympd_debug.txt.gz)

**Reply by jcorporation** [reply]
I see the mympd_scrobble trigger in the logs that sets the stickers. But this trigger executes no script. Sure you assigned the script to the scrobble trigger?

Please attach configuration.

**Comment by john-** [comment]
Ugh.  I did not properly define the script.  I selected it from the drop-down that appears when the cloud icon is clicked.  However, I did not do the now obvious thing of selecting the plus sign next to the selected script.  I selected Save first.

After properly defining he script it worked straight away.

Thanks for your patience on this.  I will close the ticket.
